Bug: Assignment to entry in nil map returned by GetValues()

**Describe the bug**
[generateSubchartAndAppDAGTasks()](https://github.com/Azure/orkestra/blob/main/pkg/workflow/templates.go#L225) is assigning value to a nil map returned by [`GetValues()`](https://github.com/Azure/orkestra/blob/main/api/v1alpha1/appgroup_types.go#L222).

**Error output:**
```shell
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 25 [running]:
testing.tRunner.func1.2(0x230d800, 0x269a0d0)
	/usr/local/go/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000186f00)
	/usr/local/go/src/testing/testing.go:1146 +0x4b6
panic(0x230d800, 0x269a0d0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/Azure/Orkestra/pkg/workflow.generateSubchartAndAppDAGTasks(0xc000161b68, 0x24c3a86, 0xc, 0xc000161b18, 0xc000161aa8, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/rajwinder/dev-tmp/orkestra/pkg/workflow/templates.go:225 +0x5c8
github.com/Azure/Orkestra/pkg/workflow.Test_generateSubchartAndAppDAGTasks(0xc000186f00)
	/Users/rajwinder/dev-tmp/orkestra/pkg/workflow/templates_test.go:61 +0x5b9
testing.tRunner(0xc000186f00, 0x257dd00)
	/usr/local/go/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x2b3
```

**To Reproduce**
Steps to reproduce the behavior:
1. Rename the attached `templates_test.go.txt` to `templates_test.go`
2. Put renamed file from step 1 into ./pkg/workflow directory.
3. Run `go test -v ./pkg/workflow/... -run Test_generateSubchartAndAppDAGTasks`

**Expected behavior**
The test in attached file should pass without any panic.

**Attached files**
[templates_test.go.txt](https://github.com/Azure/orkestra/files/6673580/templates_test.go.txt)

**Environment (please complete the following information):**
NA

**Additional context**
Currently, I'm writing unit tests for the workflow package and I found this bug.

--- a/api/v1alpha1/appgroup_types.go
+++ b/api/v1alpha1/appgroup_types.go
@@ -220,7 +220,7 @@ type ApplicationGroupStatus struct {
 // GetValues unmarshals the raw values to a map[string]interface{} and returns
 // the result.
 func (in *Application) GetValues() map[string]interface{} {
-	var values map[string]interface{}
+	values := make(map[string]interface{})
 	if in.Spec.Release.Values != nil {
 		_ = json.Unmarshal(in.Spec.Release.Values.Raw, &values)
 	}

